### PR TITLE
Widget api

### PR DESCRIPTION
**Pull request by balmas**
These changes add a javscript api to the Arethusa widget.

The following api calls are supported:

getMorph(sentenceId,wordId) - returns morphology for a token (in the same format as retrieved by the BSPMorphRetriever)

nextChunk() - navigates to the next chunk (sentence)

prevChunk() - navigates to the previous chunk (sentence)

gotoRef(sentenceId) - navigates to a specific chunk (sentence) id

refreshView() - rerenders the tree view

I think there is still more work to do here, particularly to see if there is a way to be sure the api isn't called until the application is fully loaded, but it generally seems to be working pretty well.

There is a corresponding change to the treebank template to use the alpheios-messaging library to call the Arethusa API.  I'll be doing a PR on that repo too, and you can see it in action using the simple client at https://alpheios-project.github.io/arethusa-demo-client.

Let me know what you think!

### CODE REVIEW

File: copy_to_g.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+cp dist/arethusa.min.js ~/workspace/Gardener/_assets/arethusa
```
Comment (zfletch) [inline copy_to_g.sh]: I really like these scripts. They make it easier to test changes in the applications that use the Arethusa Widget. The one thing I'm not sure about is hardcoding `~/workspace`.

What do you think about either having them copy to some directory in this repository (e.g. `./dist_widget`), taking a command line argument for the location, or using some other kind of configuration?

File: bridget_arethusa_npm_list [hunk 2]
```diff
@@ -0,0 +1,1552 @@
+arethusa@0.2.5 /home/balmas/workspace/arethusa
```
[the remark this answers is not in the document]
Comment (balmas) [reply bridget_arethusa_npm_list]: oops, sorry didn't mean to commit this

File: copy_to_g.sh (hunk 1 above)
Comment (balmas) [reply copy_to_g.sh]: oops didn't mean to commit these scripts but yes, they do make it easier to test. Your suggestion is a good one. I can make that change.

File: app/js/arethusa.core/services/api.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+     */
+    this.prevSentence = function() {
+      navigator.prevChunk();
+    };
+
+    /**  
+     * navigates application state to supplied sentenceId
+     * @param {String} sentenceId 
+     */
+    this.gotoSentence = function(sentenceId) {
```
[the remark this answers is not in the document]
Comment (balmas) [reply app/js/arethusa.core/services/api.js]: Alpheios uses the getMorph and refreshView Api calls. I can remove next/prev calls although I could envision uses for them in the future.

File: copy_to_g.sh (hunk 1 above)
Comment (zfletch) [reply copy_to_g.sh]: Well I'm glad you did commit them. It would have taken me longer to get it working without them.

Comment (balmas) [reply copy_to_g.sh]: ok, there is now a deploy_widget.sh script which will copy the files to a directly passed to the script, or else to dist_widget directory

**Pull request merged**